nntrainer/include/neuralnet.h

```diff
@@ -321,6 +321,8 @@ class NeuralNetwork : public ml::train::Model {
 
   unsigned int epochs; /**< Maximum Epochs */
 
+  unsigned int epoch_idx; /**< Number of iterations trained */
```

Comment:
You are mixing up epochs and iterations here in naming and its description.
Please keep them separate. 
Epochs count the total number of times the whole dataset is looped over.
Iterations count the total number of batches given to the model (this must NOT reset at the end of each epoch).

Reply:
Got it. Thanks for the clarification. I'll fix it as per the comment.

Don't we need to save the epoch_idx though? I think it is natural that when the user stops training, we should save the `epoch_idx` and `iteration` and start from the epoch.

Reply:
Yes, that's why `iter` member variable in `NeuralNetwork` is being saved.